activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb

```diff
@@ -599,6 +599,19 @@ def drop_table(table_name, **options)
       #  # Ignores the method call if the column exists
       #  add_column(:shapes, :triangle, 'polygon', if_not_exists: true)
       def add_column(table_name, column_name, type, **options)
+        add_column_def = build_add_column_definition(table_name, column_name, type, **options)
+        return unless add_column_def
+
+        execute(add_column_def.ddl)
+      end
+
+      def add_columns(table_name, *column_names, type:, **options) # :nodoc:
+        column_names.each do |column_name|
+          add_column(table_name, column_name, type, **options)
+        end
+      end
+
+      def build_add_column_definition(table_name, column_name, type, **options)
```

Comment:
Question is the same here for this PR, if apps should consume this, let's document it, otherwise no-doc it.